contracts/starknet/ProofReserve.cairo

```diff
@@ -162,40 +139,68 @@ func post_data_l2{
             public_key,
         );
     }
-    //todo update the root, (format, address/balance)
-    // let new_info = info.write(public_key, DataInfo(public_key=address_owner_little, balance=balance_little));
-    // tempvar arr: DataInfo* = cast(
-    //     new(DataInfo(public_key=address_owner_little, balance=balance_little)), DataInfo*);
-
-    let (timestamp) = get_block_timestamp();
-    tempvar arr: DataInfo = DataInfo(publisher= public_key, public_key=address_owner_little, asset=asset_name_little, balance=balance_little, timestamp=timestamp);
-    create_root(arr);
-    return (timestamp=timestamp);
+    let (local proofs: felt*) = alloc();
+    assert proofs[0] = address_owner_little;
+    assert proofs[1] = asset_name_little;
+    assert proofs[2] = balance_little;
+    assert proofs[3] = timestamp;
+
+    let proofs_len = 4;
+    let (root_) = calc_hash(0, proofs_len, proofs);
```

Comment:
instead of defining `proofs` like that, you can do in one-line:

```cairo
let (root_) = calc_hash(0, 4, new (address_owner_little, asset_name_little, balance_little, timestamp));
```